core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/scatterplot/ScatterplotOpDesc.java

```diff
@@ -0,0 +1,68 @@
+package edu.uci.ics.texera.workflow.operators.visualization.scatterplot;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle;
+import edu.uci.ics.texera.workflow.common.metadata.InputPort;
+import edu.uci.ics.texera.workflow.common.metadata.OperatorGroupConstants;
+import edu.uci.ics.texera.workflow.common.metadata.OperatorInfo;
+import edu.uci.ics.texera.workflow.common.metadata.OutputPort;
+import edu.uci.ics.texera.workflow.common.metadata.annotations.AutofillAttributeName;
+import edu.uci.ics.texera.workflow.common.operators.OneToOneOpExecConfig;
+import edu.uci.ics.texera.workflow.common.tuple.schema.Attribute;
+import edu.uci.ics.texera.workflow.common.tuple.schema.AttributeType;
+import edu.uci.ics.texera.workflow.common.tuple.schema.Schema;
+import edu.uci.ics.texera.workflow.operators.visualization.VisualizationConstants;
+import edu.uci.ics.texera.workflow.operators.visualization.VisualizationOperator;
+import static java.util.Collections.singletonList;
+import static scala.collection.JavaConverters.asScalaBuffer;
+
+/**
+ * Scatterplot operator to visualize the result as a scatterplot
+ * This is the description of the operator
+ */
+
+public class ScatterplotOpDesc extends VisualizationOperator {
+    @JsonProperty(value = "X-Column", required = true)
+    @AutofillAttributeName
+    public String xColumn;
+
+    @JsonProperty(value = "Y-Column", required = true)
+    @AutofillAttributeName
+    public String yColumn;
+
+    @JsonProperty(defaultValue = "false")
+    @JsonSchemaTitle("Geometric")
+    @JsonPropertyDescription("plot on a map")
+    public boolean isGeometric;
+
+    @Override
+    public String chartType() {
+        if(isGeometric)
+            return VisualizationConstants.SPATIAL_SCATTERPLOT;
+        return VisualizationConstants.SCATTERPLOT;
+    }
+
+    @Override
+    public OneToOneOpExecConfig operatorExecutor() {
+        return new OneToOneOpExecConfig(operatorIdentifier(), worker -> new ScatterplotOpExec(this));
+    }
+
+    @Override
+    public OperatorInfo operatorInfo() {
+        return new OperatorInfo(
+                "Scatterplot",
+                "View the result in a scatterplot",
+                OperatorGroupConstants.VISUALIZATION_GROUP(),
+                asScalaBuffer(singletonList(new InputPort("", false))).toList(),
+                asScalaBuffer(singletonList(new OutputPort(""))).toList());
+    }
+
+    @Override
+    public Schema getOutputSchema(Schema[] schemas) {
+        return Schema.newBuilder().add(
+                new Attribute("xColumn", AttributeType.STRING),
```

Comment:
just wonder if the output type can be tightened to `Float`?

Reply:
no, the sample data I have doesn't work with float

Reply:
Isn't all input output data in float format? I can't think of a counter example...

Reply:
sorry I am talking about the AttributeType here. The AttributeType describes the values of a column, not the name of the column... Can the values be other than `AttributeType.FLOAT`?

Reply:
you are right, I have changed it to Double. Will reflect the change in a new commit with the test cases